src/reference/asciidoc/kafka.adoc

```diff
@@ -369,6 +415,8 @@ In that case, the `ErrorHandler` will be invoked, if configured, or logged other
 
 When using `@KafkaListener`, set the `RetryTemplate` (and optionally `recoveryCallback`) on the container factory and the listener will be wrapped in the appropriate retrying adapter.
 
+A retry adapter is not provided for any of the batch <<message-listeners, message listeners>>.
```

Comment:
Because there's no easy way to determine the retry point - only the actual listener can determine that - he needs to implement retry internally.

Let's say the list has 10 records; 5 are successful - how would an adapter know that?